ChessTypeProvider/Chess.fs

```diff
@@ -66,25 +66,87 @@ type Game =
         else
             None
     member private this.GetAvailableMoves(rank, file) =
+        let checkMove side x y =
```

Comment:
First of all thanks very much for your pull-request!
However I think we can find a better way of returning the sequence of moves than using a reference cells here. For instance I think using a recursive sequence would be a good solution. Do you want to try making that change?